File: solutions/1828. Queries on Number of Points Inside a Circle/1828.py
# Time complexity: O(mn)
# Space complexity: O(m)
class Solution:
  def countPoints(
    self, 
    points: List[List[int]], 
    queries: List[List[int]]
  ) -> List[int]:
    ans = []

    for xj, yj, rj in queries:
      count = 0
      for xi, yi in points:
        if (xi - xj)**2 + (yi - yj)**2 <= rj**2:
          count += 1
      ans.append(count)

    return ans

File: solutions/1828. Queries on Number of Points Inside a Circle/1828.rs
// Time complexity: O(mn)
// Space complexity: O(m)
impl Solution {
    pub fn count_points(points: Vec<Vec<i32>>, queries: Vec<Vec<i32>>) -> Vec<i32> {
        queries
            .iter()
            .map(|query| {
                let (xj, yj, rj) = (query[0], query[1], query[2]);
                points
                    .iter()
                    .filter(|point| {
                        let (xi, yi) = (point[0], point[1]);
                        (xi - xj).pow(2) + (yi - yj).pow(2) <= rj.pow(2)
                    })
                    .count() as i32
            })
            .collect()
    }
}
